### DISCUSSION TITLE
mysql password contains '@' characters, an error occurs

### DISCUSSION THREAD

**Discussion by Leo9866** [juicedata/juicefs#2534] [General]
**What happened**:
If the password contains the '@' character, it should be able to link to the database normally.

e.g: `mysql://juicefs:#$_B8d4IPJ$g*@81c@(x.x.x.x:3306)/juicefs`
the passworld is :` #$_B8d4IPJ$g*@81c`

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Configure a database password to include the '@' character, which conflicts with the '@' character of the mysql link separator
**Anything else we need to know?**

Perhaps at line 368 in the interface.go  file, the code should be changed
`atIndex := strings.LastIndex(uri, "@")`

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:   1.0.0
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):    
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

**Comment by zhijian-pro** [comment]
Special characters maybe need to be escaped using [URL encode](https://www.w3schools.com/tags/ref_urlencode.ASP).

**Comment by Leo9866** [comment]
Ah！'@' can be defined as a special character in a string ？

**Comment by zhijian-pro** [comment]
The mysql url format is `mysql://<username>[:<password>]@(<host>:3306)/<database-name>`,`@` is a keep characters.

**Comment by Leo9866** [comment]
em，i’m get .
